src/fides/api/service/connectors/dynamic_erasure_email_connector.py

```diff
@@ -0,0 +1,258 @@
+            },
+        )
+        if not email_recipient_data:
+            logger.error(
+                "Custom request field lookup yielded no results. Skipping email send for privacy request with id {}.",
+                privacy_request.id,
+            )
+            return None
+
+        if len(email_recipient_data) > 1:
+            logger.error(
+                "Custom request field lookup yielded multiple results. Skipping email send for privacy request with id {}.",
+                privacy_request.id,
+            )
+            return None
+
+        return email_recipient_data[0][field_name]
+
+    def batch_email_send(self, privacy_requests: Query) -> None:
+        # Import here to avoid circular import
+        from fides.api.service.connectors import get_connector
+
+        logger.debug(
+            "Starting batch_email_send for connector: {} ...", self.configuration.name
+        )
+
+        db: Session = Session.object_session(self.configuration)
+
+        dataset_reference = self.config.recipient_email_address
+        dataset_key = dataset_reference.dataset
+
+        # We get the DatasetConfig instance using the dataset key provided in the configuration
+        dataset_config: Optional[DatasetConfig] = DatasetConfig.get_by(
+            db, field="fides_key", value=dataset_key
+        )
+
+        if not dataset_config:
+            logger.error(
+                "DatasetConfig with key '{}' not found. Skipping erasure email send for connector: '{}'.",
+                dataset_key,
+                self.configuration.name,
+            )
+            return
+
+        # Get the graph dataset from the dataset config
+        graph_dataset: GraphDataset = dataset_config.get_graph()
+
+        # Build the CollectionAddress using the dataset name and the collection name
+        split_field = dataset_reference.field.split(".")
+        collection_name = split_field[0]
+        collection_address = CollectionAddress(graph_dataset.name, collection_name)
+        # And get the field name from the datasetreference
+        field_name = ".".join(split_field[1:])
+
+        # Get the corresponding ConnectionConfig instance
+        connection_config: ConnectionConfig = (
+            db.execute(
+                db.query(ConnectionConfig).filter(
+                    ConnectionConfig.id == dataset_config.connection_config_id
+                )
+            )
+            .scalars()
+            .first()
+        )
+
+        # Instatiate the ConnectionConfig's connector so that we can execute the query
+        # to retrieve the email addresses based on the custom request fields
+        connector = get_connector(connection_config)
+
+        skipped_privacy_requests: List[str] = []
+        # We'll batch identities for each email address
+        # so we'll only send 1 email to each email address
+        batched_identities: Dict[str, List[str]] = {}
```

Comment:
in the generic erasure email connector, batched_identities is a list. Here we have a dictionary that maps the recipient email address to its identities